src/plugins/home/server/services/sample_data/routes/install.ts

```diff
@@ -53,21 +53,51 @@ const insertDataIntoIndex = (
   ) => Promise<any>,
   logger: Logger
 ) => {
+  // Helper function to get a nested field by path
+  function getNestedField(doc: any, path: string) {
+    // First check if the exact path exists as a field
+    if (path in doc) {
+      return doc[path];
+    }
+    // If not, treat it as a nested path
+    return path
+      .split('.')
+      .reduce((obj, key) => (obj && obj[key] !== 'undefined' ? obj[key] : undefined), doc);
+  }
+
+  // Helper function to set a nested field by path
+  function setNestedField(doc: any, path: string, value: any) {
```

Comment:
Shall we separate these `setNestedField` and `getNestedField` into utils file and add unit tests for the missing lines?

Reply:
Moved to utils: b01ec4996a35fbf4330c9e861eb3d9de0e0f9ddd